helm/tenant/templates/tenant.yaml

```diff
@@ -47,129 +47,129 @@ spec:
+  certConfig: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  podManagementPolicy: {{ dig "podManagementPolicy" "Parallel" . }}
-  {{- with (dig "readiness" (dict) .) }}
-  readiness: {{- toYaml . | nindent 4 }}
+  podManagementPolicy: {{ dig "podManagementPolicy" "Parallel" .tenant }}
+  {{- with (dig "readiness" (dict) .tenant) }}
+  readiness: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- with (dig "liveness" (dict) .) }}
-  liveness: {{- toYaml . | nindent 4 }}
+  {{- with (dig "liveness" (dict) .tenant) }}
+  liveness: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- with (dig "startup" (dict) .) }}
-  startup: {{- toYaml . | nindent 4 }}
+  {{- with (dig "startup" (dict) .tenant) }}
+  startup: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- with (dig "exposeServices" (dict) .) }}
-  exposeServices: {{- toYaml . | nindent 4 }}
+  {{- with (dig "exposeServices" (dict) .tenant) }}
+  exposeServices: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- if dig "serviceAccountName" "" . }}
-  serviceAccountName: {{ dig "serviceAccountName" "" . }}
+  {{- if dig "serviceAccountName" "" .tenant }}
+  serviceAccountName: {{ dig "serviceAccountName" "" .tenant }}
   {{- end }}
-  prometheusOperator: {{ dig "prometheusOperator" "false" . }}
-  {{- with (dig "logging" (dict) .) }}
-  logging: {{- toYaml . | nindent 4 }}
+  prometheusOperator: {{ dig "prometheusOperator" "false" .tenant }}
+  {{- with (dig "logging" (dict) .tenant) }}
+  logging: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- with (dig "serviceMetadata" (dict) .) }}
-  serviceMetadata: {{- toYaml . | nindent 4 }}
+  {{- with (dig "serviceMetadata" (dict) .tenant) }}
+  serviceMetadata: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- with (dig "env" (list) .) }}
-  env: {{- toYaml . | nindent 4 }}
+  {{- with (dig "env" (list) .tenant) }}
+  env: {{- toYaml .tenant | nindent 4 }}
   {{- end }}
-  {{- if dig "priorityClassName" "" . }}
-  priorityClassName: {{ dig "priorityClassName" "" . }}
+  {{- if dig "priorityClassName" "" .tenant }}
+  priorityClassName: {{ dig "priorityClassName" "" .tenant }}
   {{- end }}
-  {{- if dig "kes" "configuration" false . }}
+  {{- if dig "kes" "configuration" false .tenant }}
   kes:
-    image: {{ .kes.image | quote }}
-    {{- with (dig "kes" "env" (list) .) }}
-    env: {{- toYaml . | nindent 4 }}
+    image: {{ .tenant.kes.image | quote }}
+    {{- with (dig "kes" "env" (list) .tenant) }}
+    env: {{- toYaml .tenant | nindent 4 }}
     {{- end }}
-    replicas: {{ .kes.replicas | int }}
+    replicas: {{ .tenant.kes.replicas | int }}
     kesSecret:
       name: "kes-configuration"
-    imagePullPolicy: {{ .kes.imagePullPolicy | quote }}
+    imagePullPolicy: {{ .tenant.kes.imagePullPolicy | quote }}
     {{- with (dig "kes" "externalCertSecret" (dict) .) }}
-    externalCertSecret: {{- toYaml . | nindent 6 }}
+    externalCertSecret: {{- toYaml .tenant | nindent 6 }}
```

Comment:
Why take `.tenant` here？

Reply:
This is because originally Values.tenant was set and had to add it almost everywhere where a `.` was. I did a 'helm install -n <namespace> <releasename> ./tenant -f values.yaml' to debug until it didn't give an error with the Chart code changes. 
```yaml
{{- with .Values.tenant }}
{{- with .Values }}
```
Hope this clarifies. Please change it where needed, I only changed it so it works 😉

Reply:
but It wrap in `with`

Reply:
No, in my PR I had to remove that. Otherwise you can't reference "Values.secrets"

Reply:
I mean it wrapped between `with` and `end` ，so `.tenant` are for what's field?

Reply:
You're right. Just tested it without .tenant for externalCertSecret. Seems to just work.